### ISSUE TITLE
Old-style octal literal types are treated as decimal literal types

### ISSUE THREAD

**Issue by sandersn** [Microsoft/TypeScript#10101]
``` ts
// @target: es3
const i: 010 = 8;
```

Actual:

```ts
const i: 010 = 8;
      ~
      Error: Type '8' is not assignable to type '10';
```

Expected:

```ts
const i: 010 = 8;
         ~~~
         Error: Octal literal types must use ES2015 syntax. Use the syntax '0o10'.
```

**Comment by DanielRosenwasser** [comment]
We should just disallow octal literal types.

**Comment by kitsonk** [comment]
Old style octal literals are disallowed in strict mode anyways and aren't actually part of ES5, only supported by convention.

I would assume ES6 octal literals (e.g. `0o010`) would be allowed?

**Comment by arusakov** [comment]
TypeScript is supporting decimal, octal and hex literal types in the master now:
```ts
// all below are correct
const x: 255 = 255;
const z: 0o377 = 255;
const y: 0xFF = 255;
```

If old style literal types are not allowed, issue can be closed.

**Comment by sandersn** [comment]
Old style literal types are still allowed with `"target": "es3"`, so the bug still exists there. I agree with Daniel that they should just be disallowed. The compiler should give an error "Octal literal types must use ES2015 syntax. Use the syntax {0o000}". Ideally, the number in braces should be the literal that caused the error translated to ES2015 syntax.
